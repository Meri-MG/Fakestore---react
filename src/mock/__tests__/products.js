import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import * as ReactRedux from 'react-redux';
import store from '../../redux/configureStore';
import Home from '../../components/HomePage';

const mockDispatch = jest.fn();

beforeAll(() => {
  ReactRedux.useDispatch = jest.fn().mockImplementation(() => mockDispatch);
});

beforeEach(() => {
  ReactRedux.useDispatch.mockClear();
});

describe('Rockets Component test', () => {
  it('Renders Rocket Page', () => {
    const app = renderer.create(
      <Provider store={store}>
        <Home />
      </Provider>,
    )
      .toJSON();
    expect(app).toMatchSnapshot();
  });
});
