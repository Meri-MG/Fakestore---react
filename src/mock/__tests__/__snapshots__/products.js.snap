// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets Component test Renders Rocket Page 1`] = `
Array [
  <select
    className="options"
    onChange={[Function]}
    onClick={[Function]}
    required={true}
  >
    <option
      value="Select a Category"
    >
      Select a Category
    </option>
    <option
      value=""
    >
       
    </option>
  </select>,
  <div
    className="products-grid"
  />,
]
`;
